pkg/proxy/proxy_test.go

```diff
@@ -362,6 +371,7 @@ func Test_RoundTrip(t *testing.T) {
 	}
 	p.fakeToken.EXPECT().AuthenticateToken(gomock.Any(), "fake-token").Return(authResponse, true, nil)
 	p.fakeRT.expUser = "a-user"
```

Comment:
Where we have an expected user (the auth returned successfully), can we also have `system:authenticated` added to the expected groups, instead of removing it from the test?